jslib/core/http/Client.ts

```diff
@@ -0,0 +1,183 @@
+declare const loris: any;
+import {Query, QueryParam} from './Query';
+import {Errors} from '../';
+
+export interface ErrorContext {
+  key: string | number; // The key that triggered the custom message (e.g., 'ApiNetworkError' or 404)
+  request: Request,
+  response?: Response,
+}
+
+/**
+ * A basic client for making HTTP requests to a REST API endpoint.
+ */
+export class Client<T> {
+  protected baseURL: URL;
+  protected subEndpoint?: string;
+  /**
+   * Function to retrieve a custom error message for a given error context.
+   */
+  public getErrorMessage: (
+    key: string | number,
+    request: Request,
+    response?: Response
+  ) => string | undefined = () => undefined;
+
+  /**
+   * Creates a new API client instance.
+   *
+   * @param baseURL The base URL for the API requests.
+   */
+  constructor(baseURL: string) {
```

Comment:
+constructor(baseURL: string) {
+  // Reproduce the old behaviour: loris.BaseURL + '/' + baseURL + '/'
+  const origin = window.location.origin;  // always https://...
+  const full = `${origin}/${baseURL}/`;
+  this.baseURL = new URL(full);
+}